### ISSUE TITLE
Errors in new version

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Pixelizm/phppi#9]
```
( ! ) Warning: Missing argument 1 for PHPPI::setThumbSize(), called in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 77 and defined in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 569
Call Stack
#   Time    Memory  Function    Location
1   0.0002  670568  {main}( )   ..\index.php:0
2   0.0032  1271264 PHPPI->loadSettings( )  ..\index.php:33
3   0.0037  1281400 PHPPI->setThumbSize( )  ..\phppi.php:77

( ! ) Notice: Undefined variable: ff in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 307
Call Stack
#   Time    Memory  Function    Location
1   0.0002  670568  {main}( )   ..\index.php:0
2   0.0038  1281360 PHPPI->initialize( )    ..\index.php:34
3   0.0066  1746504 PHPPI->getDir( )    ..\phppi.php:1262
4   0.0066  1746632 PHPPI->getDirData( )    ..\phppi.php:207
My Gallery - /

( ! ) Notice: Undefined variable: ff in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 307
Call Stack
#   Time    Memory  Function    Location
1   0.0002  670568  {main}( )   ..\index.php:0
2   0.0038  1281360 PHPPI->initialize( )    ..\index.php:34
3   0.0104  1757864 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\template.php' )    ..\phppi.php:1281
4   0.0105  1757864 PHPPI->showPage( )  ..\template.php:12
5   0.0107  1768944 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\pages\folder.php' 
)   ..\phppi.php:991
6   0.0108  1769280 PHPPI->showGallery( )   ..\folder.php:15
7   0.0109  1769568 PHPPI->getDirData( )    ..\phppi.php:789

( ! ) Notice: Undefined variable: fd in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 308
Call Stack
#   Time    Memory  Function    Location
1   0.0002  670568  {main}( )   ..\index.php:0
2   0.0038  1281360 PHPPI->initialize( )    ..\index.php:34
3   0.0104  1757864 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\template.php' )    ..\phppi.php:1281
4   0.0105  1757864 PHPPI->showPage( )  ..\template.php:12
5   0.0107  1768944 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\pages\folder.php' 
)   ..\phppi.php:991
6   0.0108  1769280 PHPPI->showGallery( )   ..\folder.php:15
7   0.0109  1769568 PHPPI->getDirData( )    ..\phppi.php:789

( ! ) Warning: shuffle() expects parameter 1 to be array, null given in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 792
Call Stack
#   Time    Memory  Function    Location
1   0.0002  670568  {main}( )   ..\index.php:0
2   0.0038  1281360 PHPPI->initialize( )    ..\index.php:34
3   0.0104  1757864 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\template.php' )    ..\phppi.php:1281
4   0.0105  1757864 PHPPI->showPage( )  ..\template.php:12
5   0.0107  1768944 require( 
'C:\wamp\www\phppi\phppi\themes\pix\standard\pages\folder.php' 
)   ..\phppi.php:991
6   0.0108  1769280 PHPPI->showGallery( )   ..\folder.php:15
7   0.0122  1769896 shuffle ( ) ..\phppi.php:792

123

2011-23-223 (3?????_
Generated by PHP Picture Index 1.1.0 in 0.0100820 seconds

```

Original issue reported on code.google.com by `Uksus...@gmail.com` on 25 Apr 2012 at 10:58

**Comment by GoogleCodeExporter** [comment]
```
They all appear to be non-fatal errors. These will show up if error reporting 
for php is enabled (such as via the ['advanced']['debug_mode'] setting. You may 
want to try turning that off if you have it on. Otherwise your php installation 
may need to have error reporting turned off. I will try to remove those errors 
in the future but currently they do not halt phppi from running.
```

Original comment by `gorounreal` on 25 Apr 2012 at 12:27
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I also habe errors with the new version. Thumbnails are not properly generated 
and for me the version is not usable (see attachment).

Best regards,
Christoph
```

Original comment by `muelli1...@gmail.com` on 26 Apr 2012 at 6:51

Attachments:
- [Zwischenablage01.jpg](https://storage.googleapis.com/google-code-attachments/phppi/issue-9/comment-4/Zwischenablage01.jpg)

**Comment by GoogleCodeExporter** [comment]
```
As I mentioned above those errors are non fatal, turn off either the 
['advanced']['debug_mode'] setting or turn off error reporting in your php 
installation. Thumbnails are not shown on folders if there are no images in the 
immediate folder, such as if the folder only contains folders but within those 
folders are images.
```

Original comment by `gorounreal` on 26 Apr 2012 at 10:02

**Comment by GoogleCodeExporter** [comment]
```
> turn off either the ['advanced']['debug_mode']
It's false, but errors shown anyway
```

Original comment by `Uksus...@gmail.com` on 27 Apr 2012 at 5:16

**Comment by GoogleCodeExporter** [comment]
```
Then you'll have to turn off error reporting in your php install (in your 
php.ini file). I'll try to make a quick update with either these errors fixed 
or to have error reporting turned off via PHPPI.
```

Original comment by `gorounreal` on 27 Apr 2012 at 2:46

**Comment by GoogleCodeExporter** [comment]
```
BUG REPORT / FIX: When the cache files expire, the script doesn't find any 
folders. I've fixed this in the phppi.php file, as the logic in the 
getDirData() function was flawed. Were should I upload the fix, if you want it?

I also changed the getDir() function to actually skip generating a new cache 
each time it was called, as it defeated the purpose of using a cache. So I 
added logic to test the cache expiry time before generating it again. It really 
sped up the script. Again, where and to whom should I upload my suggested fixed?

```

Original comment by `isocket...@gmail.com` on 11 Jun 2012 at 9:13

**Comment by GoogleCodeExporter** [comment]
```
The setThumbSize() function call in line 77 failed to include the default 
thumbnail setting from the settings files. I fixed this warning by pulling in 
the default setting and passing that as the argument for this function call.
```

Original comment by `isocket...@gmail.com` on 11 Jun 2012 at 9:17

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the fixes, i'd be happy to check them out and update as needed. If 
you want to send them to me just zip it up and send to contact@pixelizm.com.
```

Original comment by `gorounreal` on 11 Jun 2012 at 9:48

**Comment by GoogleCodeExporter** [comment]
```
I'm currently having the same problem (when the cache files expire, the script 
doesn't find any folders). Right now I need to dissable the file_cache... :(
I also noticed that when the cache.xml is not expired, it is recreated once the 
folder is opened.
How did you fix the problem? can you send me the phppi.php file by email? 
Thank you in advance.
Regards,
Alfredo.
```

Original comment by `alfredof...@gmail.com` on 11 Jul 2012 at 10:26

**Comment by GoogleCodeExporter** [comment]
```
Both issues should now be fixed with 1.1.1. Sorry for the delay.
```

Original comment by `gorounreal` on 21 Aug 2012 at 2:09
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
thanks for both fix!
One warning shown anyway :)
( ! ) Warning: Missing argument 1 for PHPPI::setThumbSize(), called in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 107 and defined in 
C:\wamp\www\phppi\phppi\includes\classes\phppi.php on line 611
```

Original comment by `Uksus...@gmail.com` on 27 Sep 2012 at 4:41
